specs/rollup-node.md

```diff
@@ -0,0 +1,431 @@
+    - [Encoding the L1 Attributes Deposited Transaction](#encoding-the-l1-attributes-deposited-transaction)
+    - [Encoding User-Deposited Transactions](#encoding-user-deposited-transactions)
+    - [Deriving all Payload Attributes of a sequencing window](#deriving-all-payload-attributes-of-a-sequencing-window)
+      - [Building individual Payload Attributes](#building-individual-payload-attributes)
+  - [From Payload Attributes to L2 Block](#from-payload-attributes-to-l2-block)
+    - [Inductive Derivation Step](#inductive-derivation-step)
+    - [Engine API Error Handling](#engine-api-error-handling)
+    - [Finalization Guarantees](#finalization-guarantees)
+  - [Whole L2 Chain Derivation](#whole-l2-chain-derivation)
+  - [L2 Output RPC method](#l2-output-rpc-method)
+    - [Output Method API](#output-method-api)
+- [Handling L1 Re-Orgs](#handling-l1-re-orgs)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# L2 Chain Derivation
+
+[l2-chain-derivation]: #l2-chain-derivation
+
+This section specifies how the [rollup driver][g-rollup-driver] derives a sequence of L2 blocks per sequencing window.
+
+Every L2 block carries transactions of two categories:
+
+- *[deposited transactions][g-deposited]*: two kinds:
+  - derived from the L1 chain: a single *[L1 attributes deposited transaction][g-l1-attr-deposit]* (always first).
+  - derived from [receipts][g-receipts]: zero or more *[user-deposited transactions][g-user-deposited]*.
+- *[sequenced transactions][g-sequencing]*: derived from [sequencer batches][g-sequencer-batch],
+  zero or more regular transactions, signed by L2 users.
+
+------------------------------------------------------------------------------------------------------------------------
+
+## From L1 Sequencing Window to L2 Payload Attributes
+
+A [sequencing window][g-sequencing-window] is a fixed number consecutive L1 blocks that a derivation step takes as
+input. The window is identified by an `epoch`, equal to the block number of the first block in the window.
+
+The derivation of the L2 chain from the L1 chain happens in steps.
+Each step adds a variable number of L2 blocks to the L2 chain, derived from the sequencing window for the given epoch.
+For epoch `N`, the sequencing window comprises L1 blocks `[N, N + SEQUENCING_WINDOW_SIZE)`.
+Note that the sequencing windows overlap.
+
+### Reading L1 inputs
+
+The rollup reads the following data from the [sequencing window][g-sequencing-window]:
+
+- Of the *first* block in the window only:
+  - L1 block attributes:
+    - block number
+    - timestamp
+    - basefee
+    - *random* (the output of the [`RANDOM` opcode][random])
+  - L1 log entries emitted for [user deposits][g-deposits], derived transactions are augmented with
+    `blockHeight` and `transactionIndex` of the transaction in L2.
+- Of each block in the window:
+  - Sequencer batches, derived from the transactions:
+    - The transaction receiver is the sequencer inbox address
```

Comment:
What is this value? I don't see any other references to it.

Reply:
This is why it's useful to have a section dedicated to types and config variables per page. (I thought you liked that too, but then we killed that idea?) It's an address in the chain config, although named slightly differently there. See https://github.com/ethereum-optimism/optimistic-specs/blob/83fc45de30e245295f496e65f6e42560d257d0b3/opnode/rollup/types.go#L44

Reply:
Yes, I like the idea very much. Just took it out of the withdrawal doc due to time constraints.  I'd be in favour of working through each document and adding something like that in.